### ISSUE TITLE
[InsteonPLM] Add feature: LED On/Off, Hidden Door Sensor 2 Groups, Extra Data from Motion Sensor

### ISSUE THREAD

**Issue by tommycw1** [openhab/openhab1-addons#5575]
The InsteonPLM binding does not provide the following features

Ability to turn on/off the LEDs on Insteon Dimmers/Switches and Keypad Dimmers/Switches

Ability to recognize the closed command from an Insteon Hidden Door Sensor configured for two group reporting

Ability to take all data reported by the motion sensor when it responds to data requests after reporting motion

I already have working code to implement these features and plan to shortly submit a pull request

**Comment by 400killer** [comment]
Just got all 35 of my devices linked and coded. Any updates on a binding update to fix the hidden door close, and motion sensor issues?

**Comment by NickWaterton** [comment]
Any updates on this? I too implemented code for this (two groups) a long time ago, in an attempt to make the door sensors more reliable. Didn’t really help, and the Insteon code is quite fragile, so I would be loathe to mess with it, as it’s not being actively maintained anymore (I don’t think so anyway).
I thought LED brightness was implemented? It shows as a property in the debug logs anyway. I admit that I don’t see it in the device properties though.

**Comment by 400killer** [comment]
I took the plunge, and moved over to Home Assistant due to my investment in insteon products. I haven't looked back. It works with fan lincs, hidden door sensors, keypads, plugs, switches, water sensors, all of it. For instance, it reliably says which door is opened on my Sonos speakers triggered from my hidden door sensors. The only thing I miss is double tapping for fast off to control other things. The current integration deosnt support fast on or off.

**Comment by NickWaterton** [comment]
That’s using Insteon-mqtt bridge right? I was thinking of trying that out with OH. I have a lot of Insteon stuff also.

**Comment by 400killer** [comment]
No, I communicate using a USB PLM. Same one I used with Openhab. Home Assistant has an integration that supports the plms, and hubs.

**Comment by NickWaterton** [comment]
Yes, you have to use the PLM, but I’m pretty sure ha uses an Insteon PLM to mqtt bridge to send and receive messages. Should be easily adaptable to OH.
I wrote the Roomba vacuum to mqtt bridge that is used in ha, but I originally wrote if for OH.
I’ll take a look at it, see if faston can be added.

**Comment by tommycw1** [comment]
I have code that does the LED ON/OFF and use it everyday in my set up, also
a few other small - infrequently used things like setting ramp rate etc.
I'm more than happy to share but never figured out hot to do a pull request
and get this main lined.  If someone else wants to do this LMK and I'll
send you my (couple lines) of code.

This code stuff does nothing to fix the issues with the Hidden door
sensors.  It an issue with timing and ignoring signals.  I'm not a good
enough programmer to fix, so I removed and replaced my insteon sensors with
zwave and they work perfectly.
